alicloud/resource_alicloud_slb_rule.go

```diff
@@ -143,69 +144,55 @@ func resourceAliyunSlbRuleRead(d *schema.ResourceData, meta interface{}) error {
 		return WrapError(err)
 	}
 
-	d.Set("name", rule.RuleName)
-	d.Set("load_balancer_id", rule.LoadBalancerId)
-	if port, err := strconv.Atoi(rule.ListenerPort); err != nil {
+	d.Set("name", object.RuleName)
+	d.Set("load_balancer_id", object.LoadBalancerId)
+	if port, err := strconv.Atoi(object.ListenerPort); err != nil {
 		return WrapError(err)
 	} else {
 		d.Set("frontend_port", port)
 	}
-	d.Set("domain", rule.Domain)
-	d.Set("url", rule.Url)
-	d.Set("server_group_id", rule.VServerGroupId)
+	d.Set("domain", object.Domain)
+	d.Set("url", object.Url)
+	d.Set("server_group_id", object.VServerGroupId)
 
 	return nil
 }
 
 func resourceAliyunSlbRuleUpdate(d *schema.ResourceData, meta interface{}) error {
-
-	d.Partial(true)
-
 	if d.HasChange("server_group_id") {
-		req := slb.CreateSetRuleRequest()
-		req.RuleId = d.Id()
-		req.VServerGroupId = d.Get("server_group_id").(string)
+		request := slb.CreateSetRuleRequest()
+		request.RuleId = d.Id()
+		request.VServerGroupId = d.Get("server_group_id").(string)
 		client := meta.(*connectivity.AliyunClient)
 		raw, err := client.WithSlbClient(func(slbClient *slb.Client) (interface{}, error) {
-			return slbClient.SetRule(req)
+			return slbClient.SetRule(request)
 		})
 		if err != nil {
-			return WrapErrorf(err, DefaultErrorMsg, d.Id(), req.GetActionName(), AlibabaCloudSdkGoERROR)
+			return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
 		}
-		addDebug(req.GetActionName(), raw)
-		d.SetPartial("server_group_id")
+		addDebug(request.GetActionName(), raw)
 	}
 
-	d.Partial(false)
-
 	return resourceAliyunSlbRuleRead(d, meta)
 }
 
 func resourceAliyunSlbRuleDelete(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*connectivity.AliyunClient)
-	req := slb.CreateDeleteRulesRequest()
-	req.RuleIds = fmt.Sprintf("['%s']", d.Id())
-	return resource.Retry(5*time.Minute, func() *resource.RetryError {
-		raw, err := client.WithSlbClient(func(slbClient *slb.Client) (interface{}, error) {
-			return slbClient.DeleteRules(req)
-		})
-		if err != nil {
-			if IsExceptedErrors(err, []string{InvalidRuleIdNotFound}) {
-				return nil
-			}
-			return resource.NonRetryableError(WrapErrorf(err, DefaultErrorMsg, d.Id(), req.GetActionName(), AlibabaCloudSdkGoERROR))
-		}
-		addDebug(req.GetActionName(), raw)
+	slbService := SlbService{client}
+	request := slb.CreateDeleteRulesRequest()
+	request.RuleIds = fmt.Sprintf("['%s']", d.Id())
 
-		client := meta.(*connectivity.AliyunClient)
-		slbService := SlbService{client}
-		if _, err := slbService.DescribeLoadBalancerRuleAttribute(d.Id()); err != nil {
-			if NotFoundError(err) {
-				return nil
-			}
-			return resource.NonRetryableError(WrapError(err))
+	raw, err := client.WithSlbClient(func(slbClient *slb.Client) (interface{}, error) {
+		return slbClient.DeleteRules(request)
+	})
+	if err != nil {
+		if !IsExceptedErrors(err, []string{InvalidRuleIdNotFound}) {
```

Comment:
I think the line 189 should be `if IsExceptedErrors(err, []string{InvalidRuleIdNotFound}) {`